### PR TITLE
[docs] Add eas build:resign command

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Closes ENG-7423

# How

By documenting how `eas build:resign` works and where is it useful to use the command.

It also fixes typo in page's description.

# Test Plan

CleanShot 2023-02-13 at 17 09 05@2x

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/app-signing/app-credentials.mdx [hunk 1]
```diff
@@ -73,3 +74,18 @@ Provisioning profiles expire after 12 months, but this won't affect apps in prod
 ### Clearing Credentials
 
 When you use the `eas credentials` command to delete your credentials, this only removes those credentials from Expo's servers, **it does not delete the credentials from Apple's perspective**. This means that to fully delete your credentials (for example, if you want a new push notification key, however, you already have two), you'll need to do so from the [Apple Developer Console](https://developer.apple.com/account/resources/certificates/list).
+
+### Resigning new credentials
+
+You can use `eas build:resign` to codesign an existing **.ipa** for iOS to a new adhoc provisioning profile. This helps reduce time when distributing internally, for example, when you have multiple test iOS devices or want to share a build with your team members.
```
Comment (amandeepmittal) [inline docs/pages/app-signing/app-credentials.mdx:80]:
```suggestion
You can use `eas build:resign` to codesign an existing **.ipa** for iOS to a new ad hoc provisioning profile. This helps reduce time when distributing internally, for example, when you have multiple test iOS devices or want to share a build with your team members.
```

File: docs/pages/app-signing/app-credentials.mdx [hunk 2]
```diff
@@ -73,3 +74,18 @@ Provisioning profiles expire after 12 months, but this won't affect apps in prod
 ### Clearing Credentials
 
 When you use the `eas credentials` command to delete your credentials, this only removes those credentials from Expo's servers, **it does not delete the credentials from Apple's perspective**. This means that to fully delete your credentials (for example, if you want a new push notification key, however, you already have two), you'll need to do so from the [Apple Developer Console](https://developer.apple.com/account/resources/certificates/list).
+
+### Resigning new credentials
+
+You can use `eas build:resign` to codesign an existing **.ipa** for iOS to a new ad hoc provisioning profile. This helps reduce time when distributing internally, for example, when you have multiple test iOS devices or want to share a build with your team members.
```
Comment (brentvatne) [inline docs/pages/app-signing/app-credentials.mdx:80]:
```suggestion
You can use `eas build:resign` to codesign an existing **.ipa** for iOS to a new ad hoc provisioning profile. This helps reduce time when distributing internally — for example, if you want to add a new test device to an existing build, you can use this command to update the provisioning profile to include the device without rebuilding the entire app from scratch.
```

File: docs/pages/app-signing/app-credentials.mdx [hunk 3]
```diff
@@ -73,3 +74,18 @@ Provisioning profiles expire after 12 months, but this won't affect apps in prod
...
+
+Running the command will ask you to select a build that you want to resign. For example, running the command in an example project shows an available build:
```
Comment (brentvatne) [inline docs/pages/app-signing/app-credentials.mdx:82]:
```suggestion
Running the command will ask you to select a build that you want to re-sign. For example, running the command in an example project shows an available build:
```

it's re-sign in a sentence but resign in a command. the convention is to remove - in command names

File: docs/pages/app-signing/app-credentials.mdx [hunk 4]
```diff
@@ -73,3 +74,18 @@ Provisioning profiles expire after 12 months, but this won't affect apps in prod
 ### Clearing Credentials
 
 When you use the `eas credentials` command to delete your credentials, this only removes those credentials from Expo's servers, **it does not delete the credentials from Apple's perspective**. This means that to fully delete your credentials (for example, if you want a new push notification key, however, you already have two), you'll need to do so from the [Apple Developer Console](https://developer.apple.com/account/resources/certificates/list).
+
+### Resigning new credentials
```
Comment (brentvatne) [inline docs/pages/app-signing/app-credentials.mdx:78]:
```suggestion
### Re-signing new credentials
```

File: docs/pages/app-signing/app-credentials.mdx [hunk 5]
```diff
@@ -73,3 +74,18 @@ Provisioning profiles expire after 12 months, but this won't affect apps in prod
...
+Running the command will ask you to select a build that you want to resign. For example, running the command in an example project shows an available build:
+
+<ImageSpotlight
+  alt="Running eas build:resign command shows a list of available builds in a project."
+  src="/static/images/eas-build/eas-build-resign.png"
+/>
+
+After selecting the build, follow the steps to log in to your Apple Developer account. When prompted **Show devices and ask me again**, you can select a new provisioning profile.
+
+Select a new device, and the command will run the EAS Build again. Note that the build triggered this time is not a full build. It reuses the artifacts from the selected build and only codesign them to the new provisioning profile. Once the process of codesign is complete, you can use this new build link to install the **.ipa** on the iOS device whose provisioning profile was selected during the process.
```
Comment (brentvatne) [inline docs/pages/app-signing/app-credentials.mdx:91]:
```suggestion
Select a new device, and the command will run the EAS Build again. Note that the build triggered this time reuses the application artifact from the selected build and codesigns it with the new provisioning profile. Once the process of codesigning is complete, you can use this new build link to install the **.ipa** on the iOS device that was added to the provisioning profile during the process.
```

**Review by brentvatne — APPROVED** [review]

**Pull request merged**